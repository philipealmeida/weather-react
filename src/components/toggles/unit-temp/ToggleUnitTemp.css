.toggle-unit-temp {
	display: inline-block;
	padding-top: 0.8rem;
	transition: var(--transition-default);
}

.toggle-unit-temp .toggle-selector {
	width: 55px;
	height: 16px;
}

.toggle-unit-temp .toggle-selector svg {
	height: 50px;
}

.toggle-unit-temp .toggle-selector .wrapper-icons {
	width: 45%;
	height: 11.5px;
	border-radius: 0.6rem;
	z-index: 5;
	position: absolute;
	left: 0;
	background-color: var(--color-dark-mode-light-blue);
}

.toggle-unit-temp .toggle-selector:before,
.toggle-unit-temp .toggle-selector:after {
	font-size: 0.8rem;
	position: absolute;
}

.toggle-unit-temp .toggle-selector:after {
	content: 'C°';
	right: 6px;
}

.toggle-unit-temp .toggle-selector:before {
	content: 'F°';
	left: 6px;
}

.toggle-unit-temp .toggle.enabled .wrapper-icons {
	transform: translateX(100%);
}