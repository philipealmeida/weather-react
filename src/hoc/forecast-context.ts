import { createContext } from "react";

export interface ForecastContextProps {
  isFahrenheit: boolean;
  setIsFahrenheit: Function;
}

/**
 * Forecast context contains all the state and
 * utilities needed for the children components,
 * such as Forecast and Daily Forecast.
 */
export const ForecastContext = createContext<ForecastContextProps>(
  {} as ForecastContextProps
);
