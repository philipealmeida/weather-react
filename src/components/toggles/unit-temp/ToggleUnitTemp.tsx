import React, { useState, useEffect, KeyboardEvent, useContext } from "react";
import { ForecastContext } from "../../../hoc/forecast-context";
import "./ToggleUnitTemp.css";
export default function ToggleUnitTemp() {
  const { isFahrenheit, setIsFahrenheit } = useContext(ForecastContext);

  const toggleState = () => {
    setIsFahrenheit(!isFahrenheit);
  };

  const handleKeyDown = (e: KeyboardEvent<HTMLLabelElement>) => {
    if (e.key === 'Enter' || e.key === " ")
      toggleState();
  }

  return (
    <label 
      className="toggle-unit-temp toggle-wrapper"
      onKeyDown={handleKeyDown}
      tabIndex={0}
      htmlFor="toggle-unit-temp"> 
      <div className={`toggle ${isFahrenheit ? "enabled" : "disabled"}`}>
        <input
          id="toggle-unit-temp"
          name="toggle"
          type="checkbox"
          onChange={toggleState}
        />
        <div className="toggle-selector">
          <div className="wrapper-icons">
          </div>
        </div>
      </div>
    </label>
  );
}
