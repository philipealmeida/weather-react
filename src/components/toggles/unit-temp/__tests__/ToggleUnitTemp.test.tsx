// Imports
import { fireEvent, render, RenderResult, screen } from '@testing-library/react';
import { ForecastContext } from '../../../../hoc/forecast-context';
import ToggleUnitTemp from '../ToggleUnitTemp';
describe('ToggleUnitTemp renders', () => {
  const mockContextValue = {
    isFahrenheit: true,
    setIsFahrenheit: () => { }
  }

  let renderResult: RenderResult;
  const doRender = () => {
    renderResult = render(
      <ForecastContext.Provider value={mockContextValue}>
        <ToggleUnitTemp />
      </ForecastContext.Provider>);
  };

  beforeEach(() => {
    doRender();
  });

  it('should render match snapshot', () => {
    expect(renderResult).toMatchSnapshot();
  });

  describe('ToggleUnitTemp user interactions', () => {

    it('toggle with click', () => {
      const label = screen.getByTestId('toggle-unit-temp');
      fireEvent.click(label);
      expect(label).toBeDefined();
    });

    it('toggle with keydown', () => {
      const label = screen.getByTestId('toggle-unit-temp');
      fireEvent.keyDown(label, { key: 'Enter', code: 13 });
      fireEvent.keyDown(label, { key: ' ', code: 32 });
      expect(label).toBeDefined();
    });

  });

  describe('ToggleUnitTemp renders ', () => {
    const mockContextValue = {
      isFahrenheit: false,
      setIsFahrenheit: () => { }
    }
    let renderResult: RenderResult;
    const doRender = () => {
      renderResult = render(
        <ForecastContext.Provider value={mockContextValue}>
          <ToggleUnitTemp />
        </ForecastContext.Provider>);
    };

    beforeEach(() => {
      doRender();
    });

    it('should render with falsy context value', () => {
      expect(renderResult).toMatchSnapshot();
    });
  });
});


