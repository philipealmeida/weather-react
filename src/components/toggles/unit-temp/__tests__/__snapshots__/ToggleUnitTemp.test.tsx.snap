// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToggleUnitTemp renders ToggleUnitTemp renders  should render with falsy context value 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <label
        class="toggle-unit-temp toggle-wrapper"
        data-testid="toggle-unit-temp"
        for="toggle-unit-temp"
        tabindex="0"
      >
        <div
          class="toggle enabled"
        >
          <input
            id="toggle-unit-temp"
            name="toggle"
            type="checkbox"
          />
          <div
            class="toggle-selector"
          >
            <div
              class="wrapper-icons"
            />
          </div>
        </div>
      </label>
    </div>
    <div>
      <label
        class="toggle-unit-temp toggle-wrapper"
        data-testid="toggle-unit-temp"
        for="toggle-unit-temp"
        tabindex="0"
      >
        <div
          class="toggle disabled"
        >
          <input
            id="toggle-unit-temp"
            name="toggle"
            type="checkbox"
          />
          <div
            class="toggle-selector"
          >
            <div
              class="wrapper-icons"
            />
          </div>
        </div>
      </label>
    </div>
  </body>,
  "container": <div>
    <label
      class="toggle-unit-temp toggle-wrapper"
      data-testid="toggle-unit-temp"
      for="toggle-unit-temp"
      tabindex="0"
    >
      <div
        class="toggle disabled"
      >
        <input
          id="toggle-unit-temp"
          name="toggle"
          type="checkbox"
        />
        <div
          class="toggle-selector"
        >
          <div
            class="wrapper-icons"
          />
        </div>
      </div>
    </label>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`ToggleUnitTemp renders should render match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <label
        class="toggle-unit-temp toggle-wrapper"
        data-testid="toggle-unit-temp"
        for="toggle-unit-temp"
        tabindex="0"
      >
        <div
          class="toggle enabled"
        >
          <input
            id="toggle-unit-temp"
            name="toggle"
            type="checkbox"
          />
          <div
            class="toggle-selector"
          >
            <div
              class="wrapper-icons"
            />
          </div>
        </div>
      </label>
    </div>
  </body>,
  "container": <div>
    <label
      class="toggle-unit-temp toggle-wrapper"
      data-testid="toggle-unit-temp"
      for="toggle-unit-temp"
      tabindex="0"
    >
      <div
        class="toggle enabled"
      >
        <input
          id="toggle-unit-temp"
          name="toggle"
          type="checkbox"
        />
        <div
          class="toggle-selector"
        >
          <div
            class="wrapper-icons"
          />
        </div>
      </div>
    </label>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
